### PR TITLE
Ambassador dashboard updates

### LINKED ISSUE #231: as ambassador with multiple communities, I should be able to see the beneficiaries list for each community
From the ambassador dashboard, the ambassador should be able to see global statistics (aka all communities) and per community.

To list beneficiaries, lets add the "active beneficiaires" card, but calling it only "beneficiaries". If the ambassador is seeing global statistics, should not be possible to see the list of beneficiaries, but if in a community, the ambassador should be able to use the "view all" on ambassadors card and jump the the beneficiary list, the same that the manager has.

To have this community filter, we need to add the dropdown that says "my communities" in the following image. Do not add the filters on the right side yet.

This will also create another change. The reported suspicious activity should behave the same way. Should show all reports from all communities if seeing global statistics, otherwise, only from that community.

As loading effect, implement some shimmers on the numbers. This can be also useful for #217 and in general for everything else.

Image

### PR DESCRIPTION
**Pull request by PauloSousapt**
- Add swr getter for the beneficiaries, 
- Add shimmer effect component, 
- Add graphql queries for comunity beneficiaries,
- Templating
- Add loading effects to community cards and claims map

This PR fixes #231 #217 

## Tests

- [ ] Tested on mobile
- [ ] Tested on dektop

This PR should be manually tested by @obernardovieira @psoaresbj @hugomarquesdev , included on reviewers.

### CODE REVIEW

**Comment by PauloSousapt** [comment]
> 1. "view all" and "my communities" is confusing. view all says 119 beneficiaries. Is this from all communities? It should show only the ambassador communities and nothing else
> 2. reported suspicious activity is not opening. I also think the number is not correct
> 3. remove "inactive beneficiaries" and "suspicious activities" cards. They don't have any info at this point
> 4. with ambassador 3, I'm ambassador of "Larpers" but can't see the beneficiary list
> 5. should it have shimmer on community page? I just see the loading. But see the shimmer on ambassador dashboard, looks perfect 👌

1 - Then we need to remove the View all option right? Right now, in the view all we see all beneficiaries from all communities and in my communities, all beneficiaries from all my communities.
2- yeah the url is missing, will update this. I'm using the api endpoint we have for the reports also i believe we don't have yet an endpoint to get the reports per community and that is why the number is always the same
3 - Will do
4 - I'm guessing here but i think only managers can access the beneficiaries page, will need to check on the code but i think that is the reason why you can't see the list
5 - We can add it, we'll need to replace the component loading because right now that loading is showing until the request is completed, so if we just add the shimmer it won't be visible.

@obernardovieira

**Comment by obernardovieira** [comment]
1. It's the ambassador dashboard, should have only information related to the user. So yes, remove the view all.
2. ask @dev-jotape
3. .
4. that's the goal with #231. `but if in a community, the ambassador should be able to use the "view all" on ambassadors card and jump the the beneficiary list, the same that the manager has`
5. that's the goal with #217. To have shimmer loading effect, and totally remove the current loading in the middle of the screen

File: src/components/ShimmerEffect.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    @keyframes placeholderShimmer {
+        0% {
+            background-position: -468px 0;
+        }
+
+        100% {
+            background-position: 468px 0;
+        }
+    }
```
Comment (hugomarquesdev) [inline src/components/ShimmerEffect.tsx:38]: sort

File: src/views/Beneficiaries/index.tsx [hunk 2]
```diff
@@ -30,16 +32,31 @@ const Beneficiaries: React.FC<{ isLoading?: boolean }> = props => {
...
+    const variables = {address: ''};
+
+    if (auth?.type?.includes(userAmbassador) && getByKey('community')) {
+        variables.address = community.contractAddress !== undefined ? community.contractAddress.toLowerCase() : '';
+    } else if (auth?.type?.includes(userManager)) {
+        variables.address = auth?.user?.manager?.community;
+
+        // const communities: any = await getCommunities({
+        //     ambassadorAddress: auth?.user?.address
+        // });
```
Comment (hugomarquesdev) [inline src/views/Beneficiaries/index.tsx:58]: is this comment needed?

File: src/views/Communities/ClaimsMap.tsx [hunk 3]
```diff
@@ -1,28 +1,53 @@
...
+    path {
+	 	fill:white;
+        animation: dash 1s alternate infinite;
+       }
+       
+       @keyframes dash {
+         to {
+            stroke: white;
+         }
+       }
```
Comment (hugomarquesdev) [inline src/views/Communities/ClaimsMap.tsx:26]: sort

File: src/components/ShimmerEffect.tsx (hunk 1 above)
Comment (PauloSousapt) [reply src/components/ShimmerEffect.tsx]: Done. Left the animations below intentionally to standout the animation to be easier to find if any tweeks are necessary.

File: src/views/Beneficiaries/index.tsx (hunk 2 above)
Comment (PauloSousapt) [reply src/views/Beneficiaries/index.tsx]: Removed

File: src/views/Communities/ClaimsMap.tsx (hunk 3 above)
Comment (PauloSousapt) [reply src/views/Communities/ClaimsMap.tsx]: Done

**Comment by PauloSousapt** [comment]
> 1. I change the community, but the number of reports don't change. And if I select to "view" with a community selected, shows from all communities
> 2. With a community select, I select to see all beneficiaries, first it jumps to beneficiaries list with no beneficiaries and then it starts loading but never loads anything.

Yes @obernardovieira, all of that is happening because i'm implementing new backend endpoints that aren't merged yet (they will need to be merge in sync, otherwise it will break the current live implementation) to fully test these changes is only possible locally, using the backend branch (list-beneficiaries) with the new endpoints.

**Review by obernardovieira — APPROVED** [review]
It's related to https://github.com/impactMarket/backend/pull/410 right?
Then, lets merge both.
@hugomarquesdev any final review?

**Review by hugomarquesdev — APPROVED** [review]

**Pull request merged**